pkg/server/plugin/nodeattestor/awsiid/eks.go

```diff
@@ -0,0 +1,300 @@
+	eks.DescribeNodegroupAPIClient
+}
+
+const (
+	eksNodeListTTL = "30s" // pull EKS node list again after 30 seconds
+	eksRetries     = 5
+)
+
+var (
+	eksNodeListDuration, _ = time.ParseDuration(eksNodeListTTL)
+)
+
+type eksValidationConfig struct {
+	EKSClusterNames []string `hcl:"eks_cluster_names"`
+}
+
+type eksValidator struct {
+	eksNodeList              map[string]any
+	eksNodeListValidDuration time.Time
+	eksConfig                *eksValidationConfig
+	mutex                    sync.RWMutex
+	// eksAccountListCacheTTL holds the cache ttl from configuration; otherwise, it will be set to the default value.
+	eksAccountListCacheTTL time.Duration
+	log                    hclog.Logger
+	// retries fix number of retries before ttl is expired.
+	retries int
+	// require for testing
+	clk clock.Clock
+}
+
+func newEKSValidationBase(config *eksValidationConfig) *eksValidator {
+	client := &eksValidator{
+		eksNodeList: make(map[string]any),
+		eksConfig:   config,
+		retries:     eksRetries,
+		clk:         clock.New(),
+	}
+
+	return client
+}
+
+func (o *eksValidator) getRetries() int {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+	return o.retries
+}
+
+func (o *eksValidator) decrRetries() int {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+	if o.retries > 0 {
+		o.retries--
+	}
+
+	return o.retries
+}
+
+func (o *eksValidator) configure(config *eksValidationConfig) error {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.eksConfig = config
+
+	// While doing configuration invalidate the map so we don't keep using old one.
+	o.eksNodeList = make(map[string]any)
+	o.retries = eksRetries
+
+	o.eksAccountListCacheTTL = eksNodeListDuration
+
+	return nil
+}
+
+func (o *eksValidator) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// IsNodeInCluster method checks if the Node ID attached on the node is part of the EKS cluster.
+func (o *eksValidator) IsNodeInCluster(ctx context.Context, eksClient EKSClient, asClient autoscaling.DescribeAutoScalingGroupsAPIClient, nodeID string) (bool, error) {
+	reValidatedCache, err := o.validateCache(ctx, eksClient, asClient)
+	if err != nil {
+		return false, err
+	}
+
+	nodeIsmemberOfCluster, err := o.lookupCache(ctx, eksClient, asClient, nodeID, reValidatedCache)
+	if err != nil {
+		return false, err
+	}
+
+	return nodeIsmemberOfCluster, nil
+}
+
+// validateCache validates cache and refresh if its stale
+func (o *eksValidator) validateCache(ctx context.Context, eksClient EKSClient, asClient autoscaling.DescribeAutoScalingGroupsAPIClient) (bool, error) {
+	isStale := o.checkIfEKSNodeListIsStale()
+	if !isStale {
+		return false, nil
+	}
+
+	// cache is stale, reload the account map
+	_, err := o.reloadNodeList(ctx, eksClient, asClient, false)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (o *eksValidator) lookupCache(ctx context.Context, eksClient EKSClient, asClient autoscaling.DescribeAutoScalingGroupsAPIClient, accountIDOfNode string, reValidatedCache bool) (bool, error) {
```

Comment:
```suggestion
func (o *eksValidator) lookupCache(ctx context.Context, eksClient EKSClient, asClient autoscaling.DescribeAutoScalingGroupsAPIClient, nodeID string, reValidatedCache bool) (bool, error) {
```
This `accountIDOfNode` parameter looks like a node ID, not an account ID?